docs/naps/4-async-slicing.md

```diff
@@ -0,0 +1,630 @@
+high priority future work. One way to implement this behavior is to emit a `slice_ready`
+signal per layer that only contains that layer's slice response.
+
+
+## Alternatives
+
+### Extend the existing experimental async code
+
+- Already being used somewhat successfully by some napari users in the wild.
+- Can still reuse some of the code, tooling, and learnings from this project.
+- The technical design of this approach was not well received (i.e. there is a reason it is experimental).
+- Does not address goal 2.
+
+### Just call `set_view_slice` or `refresh` asynchronously
+
+- Simple to implement with few code changes needed.
+- Needs at least one lock to provide sensible access to layer slice state.
+- This will need to be acquired to access this state and at the beginning of many methods that access any of that state.
+- How to emit events on the main thread?
+- Does not address goal 2.
+    
+### Just access `data` asynchronously
+
+- Targets main cause of unresponsiveness (i.e. reading data).
+- No events are emitted on the non-main thread.
+- Less lazy when cancelling is possible (i.e. we do more work on the main thread before submitting the async task).
+- Splits up slicing logic into pre/post data reading, making program flow harder to follow.
+- Does not address goal 2.
+    
+### Use `QThread` and similar utilities instead of `concurrent.futures`
+
+- Standard way for plugins to support long running operations.
+- Can track progress and allow more opportunity for cancellation with `yielded` signal.
+- Can easily process done callback (which might update Qt widgets) on main thread.
+- Need to define our own task queue to achieve lazy slicing.
+- Need to connect a `QObject`, which ties our core to Qt, unless the code that controls threads does not live in core.
+    
+### Use `asyncio` package instead of `concurrent.futures`
+
+- May improve general readability of async code for some.
+- Mostly syntactic sugar on top of `concurrent.futures`.
+- Likely need an `asyncio` event loop distinct from Qt's main event loop, which could be confusing and cause issues.
+
+
+## Discussion
+
+- [Initial announcement on Zulip](https://napari.zulipchat.com/#narrow/stream/296574-working-group-architecture/topic/Async.20slicing.20project).
+    - Consider (re)sampling instead of slicing as the name for the operation discussed here.  
+- [Problems with `NAPARI_ASYNC=1`](https://forum.image.sc/t/even-with-napari-async-1-data-loading-is-blocking-the-ui-thread/68097/4)
+    - The existing experimental async code doesn't handle some seemingly simple usage.
+- [Remove slice state from layer](https://github.com/napari/napari/issues/4682)
+    - Feature request to remove slice/render state from the layer types.
+    - Motivated by creating multiple slices of the same layers for multiple canvases.
+    
+### Open questions
+
+- Should `Dims.current_step` represent the last slice position request or the last slice response?
```

Comment:
I lean towards request - as from API you could then set `dims.current_step = ...` to trigger a request, but open to other thoughts

Reply:
That's how it is in the current prototype. It implies that mutating that particular model state (and any other similar ones), triggers an async behavior.